packages/core/src/scripts/migrate-legacy-project.ts

```diff
@@ -0,0 +1,243 @@
+import path from 'path';
+import { promises as fs } from 'fs';
+import type { ManifestData, ImplDeployment } from '../manifest';
+import type { StorageItem, StorageLayout, TypeItem } from '../storage';
+
+const OPEN_ZEPPELIN_FOLDER = '.openzeppelin';
+const MIGRATION_FILE_LOCATION = 'openzeppelin-cli-migration.json';
+
+export async function migrateLegacyProject(): Promise<void> {
+  const manifestList = await findManifests();
+  const { migrations, migrationData } = await migrateManifestFiles(manifestList);
+
+  for (const manifestFile in migrations) {
+    const newManifest = migrations[manifestFile];
+    await writeJSONFile(getNewManifestLocation(manifestFile), newManifest);
+  }
+
+  await writeJSONFile(MIGRATION_FILE_LOCATION, migrationData);
+}
+
+async function migrateManifestFiles(manifestList: string[]): Promise<MigrationOutput> {
+  const migrationData: Record<string, unknown> = {};
+  const migrations: Record<string, ManifestData> = {};
+
+  for (const manifestFile of manifestList) {
+    const oldManifest = JSON.parse(await fs.readFile(manifestFile, 'utf8'));
+    const { manifestVersion } = oldManifest;
+
+    if (manifestVersion === undefined) {
+      continue;
+    }
+
+    const [major] = manifestVersion.split('.');
+
+    if (major !== '2') {
+      continue;
+    }
+
+    const network = getNetworkName(manifestFile);
+    migrationData[network] = transformProxies(oldManifest.proxies);
+    migrations[manifestFile] = updateManifest(oldManifest);
+  }
+
+  return {
+    migrations,
+    migrationData,
+  };
+}
+
+async function findManifests(): Promise<string[]> {
+  const files = await fs.readdir(OPEN_ZEPPELIN_FOLDER);
+  return files.filter(isManifestFile).map(preppendPath);
+}
+
+function preppendPath(location: string): string {
+  return path.join(OPEN_ZEPPELIN_FOLDER, location);
+}
+
+function isManifestFile(fileName: string): boolean {
+  const network = getNetworkName(fileName);
+  return isTestnet(network) || isDevnet(network);
+}
+
+function getNetworkName(fileName: string): string {
+  return path.basename(fileName, '.json');
+}
+
+function isDevnet(network: string) {
+  return network.includes('dev');
```

Comment:
This is too permissive IMO. We should use a regular expression
```suggestion
  return /^dev-\d+$/.test(network);
```